### ISSUE TITLE
Adopt JSDoc for documenting our code

### ISSUE THREAD

**Issue by RubenSmn** [Greenstand/treetracker-web-map-client#1662]
# Proposition

Adopt [JSDoc](https://jsdoc.app/about-getting-started.html) for our code documentation.

## Why

We're using a lot of custom components, it is not always clear what kind of props the component expect us to pass or what types the props need to be.

I think this could be a good alternative to something like TypeScript, where we would have to change the entire code base. Using JSDoc we can incrementally update our documentation of our custom components and utility functions. This way we can still get the benefits of auto completion and such.

---

Some hints:

- Please read our [readme](https://github.com/Greenstand/treetracker-web-map-client#treetracker-web) for more information/guide/tutorial.
- For design guidelines checkout our [Figma design](https://www.figma.com/file/XdYFdjlsHvxehlrkPVYq0l/Greenstand-Webmap).
- Here is [an engineering book](https://greenstand.gitbook.io/engineering/) in Greenstand.
- To know more about our organization, visit our [website](https://greenstand.org).
- If you want to join the slack community (some resources need the community member's permission), please leave your email address.

**Comment by RubenSmn** [comment]
@dadiorchen @khalatevarun what are you guys' opinions about this?

**Comment by dadiorchen** [comment]
I'm good with this, so this is a way we can use autocompletion without TS?

**Comment by elidakirigo** [comment]
hi, I'd like to contribute to the documentation of this project.

**Comment by khalatevarun** [comment]
Is there any particular reason to not adopt typescript in our codebase? @dadiorchen @RubenSmn

**Comment by dadiorchen** [comment]
@khalatevarun maybe we can consider typescript at the beginning of the app, but now I don't think it worth the effort to do a big refactor.

**Comment by RubenSmn** [comment]
I agree with Dadior, the refactor would a thing we would have to do in a single change. With JSDoc we can incrementally document our code.
